### PR TITLE
Added __waitFor action error types (closes #257)

### LINKED ISSUE #257: TestCafe server throws an exception if globalWaitFor failed
When globalWaitFor is failed, there is no err.relatedSourceCode, so server throws an exception in `getStepCode`.

### PR DESCRIPTION
**Pull request by georgiy-abbasov**
\cc @inikulin, @AlexanderMoskovkin, @VasilyStrelyaev
PR

### CODE REVIEW

File: src/reporter/errors/templates.js [hunk 1]
```diff
@@ -261,6 +261,30 @@ export default {
         ${getScreenshotInfoStr(err)}
     `,
 
+    [TYPE.incorrectGlobalWaitForActionEventArgument]: err => dedent`
+        ${getMsgPrefix(err, CATEGORY.actionError)}Error at step <span data-type="step-name">${err.stepName}</span>:
+
+        <code data-type="api">__waitFor</code> action's first parameter should be a function.
```
Comment (VasilyStrelyaev) [inline src/reporter/errors/templates.js]: does anybody think it looks prettier with a colon?
```
        <code data-type="api">__waitFor</code>: action's first parameter should be a function.
```

File: src/reporter/errors/templates.js [hunk 2]
```diff
@@ -261,6 +261,30 @@ export default {
...
+
+        ${getScreenshotInfoStr(err)}
+    `,
+
+    [TYPE.incorrectGlobalWaitForActionTimeoutArgument]: err => dedent`
+        ${getMsgPrefix(err, CATEGORY.actionError)}Error at step <span data-type="step-name">${err.stepName}</span>:
+
+        <code data-type="api">__waitFor</code> action's "timeout" parameter should be a positive number.
```
Comment (VasilyStrelyaev) [inline src/reporter/errors/templates.js]: ...and the same here?
```
        <code data-type="api">__waitFor</code>: action's "timeout" parameter should be a positive number.
```

File: src/reporter/errors/templates.js [hunk 3]
```diff
@@ -261,6 +261,30 @@ export default {
...
+
+        ${getScreenshotInfoStr(err)}
+    `,
+
+    [TYPE.globalWaitForActionTimeoutExceeded]: err => dedent`
+        ${getMsgPrefix(err, CATEGORY.timeout)}Error at step <span data-type="step-name">${err.stepName}</span>:
+
+        <code data-type="api">__waitFor</code> user code timeout exceeded.
```
Comment (VasilyStrelyaev) [inline src/reporter/errors/templates.js]:
```
        <code data-type="api">__waitFor</code>: user function's timeout exceeded.
```

Comment (AlexanderMoskovkin) [inline src/reporter/errors/templates.js]: @VasilyStrelyaev Maybe we should use [the same description](https://github.com/DevExpress/testcafe/pull/267/files#diff-4334bbbf0649f8c98cd2becdc41b7599L259) as for the `waitFor` action? They do the same and have the same arguments. The only difference - they are used in different places: `waitFor` in steps only via `act.waitFor`, but `__waitFor` in any user's script (test, test helpers or mixins) as a global variable

Comment (AlexanderMoskovkin) [inline src/reporter/errors/templates.js]: And we have this approach (without `:`) in all other error descriptions: https://github.com/georgiy-abbasov/testcafe-phoenix/blob/implement-global-wait-error-types/src/reporter/errors/templates.js

Comment (VasilyStrelyaev) [inline src/reporter/errors/templates.js]: yep, so
```
         <code data-type="api">__waitFor</code> action's timeout exceeded.
```

**Comment by georgiy-abbasov** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by inikulin** [comment]
lgtm

**Pull request merged**